x/mongo/driver/batch_cursor.go

```diff
@@ -326,6 +329,40 @@ func calcGetMoreBatchSize(bc BatchCursor) (int32, bool) {
 	return gmBatchSize, true
 }
 
+// commentToBSONCoreValue will convert the input comment from an "any" type to
+// a bsoncore.Value. If the conversion cannot occur, this function returns nil.
+// If marshaling the comment for the bsoncore.Value Data fails, this comment
+// will return an error.
+func commentToBSONCoreValue(comment interface{}) (*bsoncore.Value, error) {
+	if comment == nil {
+		return nil, nil
+	}
+
+	// If the comment is already a bsoncore.Value, then do nothing.
+	if value, ok := comment.(bsoncore.Value); ok {
+		return &value, nil
+	}
+
+	refValue := reflect.ValueOf(comment)
+	if refValue.Kind() == reflect.Map && refValue.Len() > 1 {
+		return nil, nil
+	}
```

Comment:
The existing logic for converting `comment` into a BSON document ([here](https://github.com/mongodb/mongo-go-driver/blob/fc7484059d836cdcfaf42f0dd9224625d3108748/mongo/collection.go#L1214) and [here](https://github.com/mongodb/mongo-go-driver/blob/fc7484059d836cdcfaf42f0dd9224625d3108748/mongo/mongo.go#L342)) allows maps of any length. We should keep the existing behavior for converting `comment` to BSON and remove this check.